### PR TITLE
Fix constraints after schema change

### PR DESCRIPTION
**Pull request by mponomar**

### CODE REVIEW

File: db/osqlcomm.c [hunk 1]
```diff
@@ -6436,6 +6436,7 @@ int osql_process_packet(struct ireq *iq, unsigned long long rqid, uuid_t uuid,
             return conv_rc_sql2blkop(iq, step, -1, ERR_NO_SUCH_TABLE, err,
                                      tablename, 0);
         }
+        iq->origdb = iq->usedb;
```
Comment (ldennis) [inline db/osqlcomm.c]: Can we instead set `iq->usedbtablevers = iq->usedb->tableversion` in constraints.c? Using `origdb` is a bit confusing to me.

Comment (akshatsikarwar) [inline db/osqlcomm.c]: 'usedtableversion' is supposed to be the version the replicant sent. Not the version currently on the master, which 'struct db' will provide.

Comment (ldennis) [inline db/osqlcomm.c]: Yes, but I think this is just a "hack" to pass the version test in record.c for cascade deletes/updates. (We do this for usedb too)

Comment (ldennis) [inline db/osqlcomm.c]: In the cases of cascade deletes/updates, replicants didnt send any versions. Unless, we make replicants check for constraints.

Comment (mponomar) [inline db/osqlcomm.c]: Neither way is particularly clean.  I'm ok with either.  If origdb is confusing, let's set usedtableversion before calling into record.c.  Either way works.

Comment (ldennis) [inline db/osqlcomm.c]: I am ok with origdb too, though I slightly prefer usedbtablevers.

Comment (ldennis) [inline db/osqlcomm.c]: FYI, I did a similar usedb-version-related fix (/hack) in #333 for the sc_timepart.test.

**Comment by mponomar** [comment]
> What where the other tests in constraints testing?
There's a constraint test there.  The failure we were seeing was due to the table with constraints being recreated several times, and thus having a different version number from the referencing table.  That's why there's a `rebuild` in the new test case.

**Pull request merged**